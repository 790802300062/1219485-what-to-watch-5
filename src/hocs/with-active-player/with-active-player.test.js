import React from "react";
import PropTypes from "prop-types";
import renderer from "react-test-renderer";
import {filmListMock} from "../../test-data/test-data";
import withActivePlayer from "./with-active-player";

const {previewImage, videoPreview} = filmListMock[0];

const mockVideoPlayerSettings = {
  previewImage,
  videoPreview,
  width: 280,
  height: 175,
  filmId: 1,
};

const MockComponent = (props) => {
  return (
    <div>
      {props.renderPlayer(mockVideoPlayerSettings)}
    </div>
  );
};

const MockComponentWrapped = withActivePlayer(MockComponent);

it(`renders withActivePlayer`, () => {
  const tree = renderer
    .create(
        <MockComponentWrapped />,
        {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

MockComponent.propTypes = {
  renderPlayer: PropTypes.func.isRequired,
};

