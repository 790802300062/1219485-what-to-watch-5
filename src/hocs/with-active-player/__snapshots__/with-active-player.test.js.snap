// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders withActivePlayer 1`] = `<div />`;
